docs/design/rosbag2_playback_time.md

```diff
@@ -0,0 +1,115 @@
+# Rosbag2 Playback Time Handling
+
+## Goal
+
+We need rosbag2 playback to do more than just playing back in real time according to the system clock.
+
+These situations must be handled
+1. No ROS-time, rosbag2 plays back messages without considering external time
+2. Rosbag2 drives ROStime by publishing to the /clock topic
```

Comment:
Does rosbag2 need to handle any conflicts in case two rosbags are playing or another node is publishing to this topic?

It doesn't seem like a common use case, but I think is something to consider.

Reply:
That's a good callout! I'll add a note for that - I think I'll just say "the user is responsible for making sure that no more than one time source is publishing to `/clock` at the same time, rosbag will provide no deconflicting". Does that sound all right?

I also just realized that setting one bag to control time and others to receive time externally, provides a way to synchronize playback between multiple separate bags! This was unintentional, but I'll call that out as well